### DISCUSSION TITLE
How to change the colors of the points, on "Editing" Example?

### DISCUSSION THREAD

**Discussion by cabezon666** [Mapsui/Mapsui#2621] [Development]
Hello,

I'm stuck trying to change the colors depending of some rules, but for example, on the "Editing" example, when I want to Add a Point, I try to change the color playing with the **edtiManager** value, but im getting stuck in change the point values.

Those points I need to move it, modify and delete. That is the reason why I need a EditLayer to create them.

thank you in advance.
![Screenshot 2024-05-07 173307](https://github.com/Mapsui/Mapsui/assets/7995000/3044b0d1-3874-4792-8aa2-a4de6cd55883)

**Comment by pauldendulk** [comment]
The purple color is determined by the EditingSample.CreateEditLayerBasicStyle() method. Changing that color will change it for all features in editing mode, because that is how the sample is currently written.

What you want is to have different colors for different features in the layer. There are two ways to do that:
- Using a ThemeStyle. This is the preferred way. It takes information from the feature to determine which style to apply. This is used in the EditingSample.CreateSelectedStyle: ```private static ThemeStyle CreateSelectedStyle() => new(f => (bool?)f["Selected"] == true ? _selectedStyle : _disableStyle);```.
- Using IFeature.Styles.Add(CreateAStyleForThisFeature()). This seems easier at first but you need to track individual features to update things later on and it gets more complicated.

So my advice rewrite the EditingSample.CreateEditLayerBasicStyle() to use a ThemeStyle and update to 'properties' of the feature to give it a distinct color.

**Reply by cabezon666** [reply]
Thank you sir for the kind reply.  I will give it a look and I will let you know.
Thank you again!
